### ISSUE TITLE
[OTHER] How to dynamic add / remove visual elements? 

### ISSUE THREAD

**Issue by failable** [Avaiga/taipy#1426]
### What would you like to share or ask?

When user select different `collection` dropdown, I'd like to display all the `columns` of that collection using multiple `input`s to accept user input values to filter and query. Since different collections have different numbers of columns, how can I dynamically adjust the numbers of `input`s ? 

Thanks.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

**Comment by FlorianJacta** [comment]
It seems like you need partials. Please read this [article](https://docs.taipy.io/en/develop/tutorials/visuals/4_partials/).

If I understand correctly you want to dynamically change your page by creating *input* visual elements.

**Comment by failable** [comment]
@FlorianJacta Hello, I want to adjust the numbers of the visual elements, not the items of a single element like `selector`.

In streamlit, it looks like 

<img width="320" alt="image" src="https://github.com/Avaiga/taipy/assets/5475262/3bf1099b-8710-4061-a62f-5b14fa1f20fe">


<img width="324" alt="image" src="https://github.com/Avaiga/taipy/assets/5475262/f63b28ee-933a-4eef-a201-4603bc1d5ae4">


<img width="311" alt="image" src="https://github.com/Avaiga/taipy/assets/5475262/eba00554-5789-4d62-9257-d2d32cddb359">

Notice how the numbers of text input elements changed when the different collections dropdown is selected.


Revalent code:

```python
def sidebar_qdrant() -> tuple[QdrantClient, str, dict | None, int]:
    qdrant_url = os.getenv("QDRANT_URL", "http://localhost:6333")
    qdrant_url = st.sidebar.text_input("Qdrant url", qdrant_url)
    qdrant_client = QdrantClient(qdrant_url)

    collections = qdrant_client.get_collections().collections
    collection_names = [x.name for x in collections]
    collection_name = st.sidebar.selectbox("Collection", collection_names)
    collection_info = qdrant_client.get_collection(collection_name)

    filterable_payload_types = ["keyword", "text"]
    payload_schema = {
        k: v
        for k, v in collection_info.payload_schema.items()
        if v.data_type in filterable_payload_types
    }
    # Fixed field ordering for better user experience.
    payload_schema = dict(
        sorted(payload_schema.items(), key=lambda x: (x[1].data_type, x[0])),
    )
    query_filter = None
    if payload_schema:
        st.sidebar.markdown("### Filters")

        filters = []
        for field, schema in payload_schema.items():
            value = st.sidebar.text_input(
                f"{field} ({str.capitalize(schema.data_type)})",
            ) # <--------------------------- !!!!!! Dynamically created text inputs
            if value:
                filters += [
                    create_filter(field, schema, value),
                ]

        if filters:
            query_filter = {
                "must": filters,
            }

    limit = st.sidebar.text_input("Limit", 10)

    return qdrant_client, collection_name, query_filter, limit
```

**Comment by FlorianJacta** [comment]
You can almost use the same code that you use in Streamlit by having a partial that you can change in real-time. Some documentation on it: https://docs.taipy.io/en/develop/tutorials/visuals/4_partials/

**Comment by failable** [comment]
Thanks for the link! I'm going to figure it out. Will comment back if I need further help.

**Comment by FlorianJacta** [comment]
The only difficult part is to bind these variables to Python variables. To do so, you can use a dictionary with all the keys you need for all your forms. Or you can dynamically change your dictionary. You have to create this *forms* variable containing every variable of your form. Also, you have to separate the logic of the page and the logic of your app interactivity.

The code below does not work; this is just to demonstrate what I mean.

```python
def sidebar_qdrant(state) -> tuple[QdrantClient, str, dict | None, int]:
    with tgb.Page() as new_partial:
        qdrant_url = os.getenv("QDRANT_URL", "http://localhost:6333")
        # tgb.input(label="Qdrant URL", "{qdrant_url}") # - create it somewhere else
        qdrant_client = QdrantClient(state.qdrant_url)
    
        collections = qdrant_client.get_collections().collections
        collection_names = [x.name for x in collections]
        collection_name = st.sidebar.selectbox("Collection", collection_names)
        collection_info = qdrant_client.get_collection(collection_name)
    
        filterable_payload_types = ["keyword", "text"]
        payload_schema = {
            k: v
            for k, v in collection_info.payload_schema.items()
            if v.data_type in filterable_payload_types
        }
        # Fixed field ordering for better user experience.
        payload_schema = dict(
            sorted(payload_schema.items(), key=lambda x: (x[1].data_type, x[0])),
        )
        query_filter = None
        if payload_schema:
            tgb.text("### Filters", mode="md")
    
            filters = []
            for field, schema in payload_schema.items():
                tgb.input(
                    value="{forms."+str(field)+"}",
                    label=f"{field} ({str.capitalize(schema.data_type)})",
                ) # <--------------------------- !!!!!! Dynamically created text inputs
                if value:
                    filters += [
                        create_filter(field, schema, value),
                    ]
    
            if filters:
                query_filter = {
                    "must": filters,
                }
    
        tgb.input("{forms.limit}", label="Limit")
    
    state.forms = payload_schema
    state.your_partial.update_content(state, new_partial)
```

**Comment by failable** [comment]
Hello, I came up with this https://gist.github.com/failable/0379edf7a5d82024a69a50194295372f

Any idea why I got two of the filter parts? The below one is not updated when I change the collection name.

Thanks.


<img width="277" alt="Screenshot 2024-06-19 at 11 18 34" src="https://github.com/Avaiga/taipy/assets/5475262/1594e4c0-5581-4d89-81ea-1eca5e9e62ac">

**Comment by FlorianJacta** [comment]
Could you rephrase? I am not sure to understand what you mean. Why is there a blue and a white part?

**Comment by failable** [comment]
> Why is there a blue and a white part?

Yes. The white part seems to be the correct one while the blue part is not.

**Comment by FlorianJacta** [comment]
Is the blue part not supposed to be there at all, or is it just incorrect? Can you do some print inside your "for"?

**Comment by failable** [comment]
> Is the blue part not supposed to be there at all

Yes

When I launch the app, the logs look like

<img width="957" alt="image" src="https://github.com/Avaiga/taipy/assets/5475262/609dea64-c49c-4c66-ae9d-b537576f2bbd">

and the page looks like

<img width="1237" alt="image" src="https://github.com/Avaiga/taipy/assets/5475262/0aa4d5e8-ea13-4460-831b-bb70b7730a9d">

---------------------
When I change the `collection` to `news`, the logs look like 
<img width="961" alt="image" src="https://github.com/Avaiga/taipy/assets/5475262/b8222be7-fb92-45a1-8393-c2e2ab0cfb9a">

and the page look like 

<img width="1240" alt="image" src="https://github.com/Avaiga/taipy/assets/5475262/3c4ad935-1f86-43d9-b9a1-bcd811401234">


The filters of the last collection is left in the blue part. And the blue part is not supposed to be there at all when the app is launched.

**Comment by FlorianJacta** [comment]
The blueprint seems to be created when the *on_init* is called. 
Try to change this 

`gui.add_partial("filter_parial")` to `gui.add_partial("")`

And try to erase the indent you have with the section about the GUI so it is at the level of your `if`.

```python
        gui = Gui(page=page)
        filter_partial = gui.add_partial("filter_partial")
        gui.run(
            title="Qdrant viewer",
            dark_mode=False,
            debug=True,
            use_reloader=True,
            port=5001,
        )
```

Then, try to put *use_reloader* to False.

**Comment by failable** [comment]
> Try to change this gui.add_partial("filter_parial") to gui.add_partial("")

Not working

> Then, try to put use_reloader to False.

Not working

> And try to erase the indent you have with the section about the GUI so it is at the level of your if.

This works!

Thanks for taking time to dig into this issue!

**Comment by FlorianJacta** [comment]
Great!
